recipe/build.sh

```diff
@@ -96,7 +97,8 @@ fi
             --enable-using-memchecker \
             --enable-static=yes \
             --enable-ros3-vfd \
-	    ${hdf5_disable_tests}
+	    ${hdf5_disable_tests} \
+	    || (cat config.log; false)
```

Comment:
This should help us see any details related to configuration errors (atm ones on macOS, but is generally useful outside of this need)